src/Filter/ModelFilter.php

```diff
@@ -140,4 +146,29 @@ private function getParentAlias(ProxyQueryInterface $query, string $alias): stri
 
         return $parentAlias;
     }
+
+    /**
+     * @param ProxyQueryInterface<object> $query
+     */
+    private function transformSearchValues(ProxyQueryInterface $query, array $values): array
```

Comment:
This implementation is a bit problematic, but it is needed since: https://github.com/ramsey/uuid-doctrine/issues/18

I know if it is not symfony/uid, but it is happening the same, doctrine has a limitation when passing an array of object to the setParameter. It tries to convert everything to the __toString representation and can't be changed.

Here I tried to iterate over those values, extract the identifier, extract the type of that identifier (assuming the primary key is only one field...) and try to convert to the database value needed to search.

It works for single PK entities, not sure what would happen for compound primary keys. Also not sure if it is something that can happen. This filter is used on an entity with a foreign key to another (that foreign key is only allowed to be on a single field, right?

wdyt @VincentLanglet ?

Reply:
Something that I might read later: https://www.doctrine-project.org/projects/doctrine-orm/en/2.14/tutorials/composite-primary-keys.html

Reply:
Another option to try mitigate the problem could be unroll the loop into multiple andWheres one for each entity in the list. Even in that case, the problem will be the same for each item. I will need to know its identifier, its type, and transform it to the database representation, because I can't pass the full object.

Reply:
I have refactored the code to be simpler but handle way more cases.